Trappist/src/Promact.Trappist.Test/Tests/TestsRepositoryTest.cs

```diff
@@ -151,6 +166,56 @@ public TestsRepositoryTest(Bootstrap bootstrap) : base(bootstrap)
             Assert.Equal(0, _trappistDbContext.Test.Count());
         }
 
+        /// <summary>
+        /// Adds the selected category to TestCategory
+        /// </summary>      
+        [Fact]
+        public async Task AddTestCategory()
+        {
+            var category = CreateCategory();
+            await _categoryRepository.AddCategoryAsync(category);
+            var test = CreateTest("Final");
+            await _testRepository.CreateTestAsync(test);
+            TestCategory testcategory = new TestCategory();
+            testcategory.TestId = test.Id;
+            testcategory.CategoryId = category.Id;
+            List<TestCategory> testCategoryList = new List<TestCategory>();
+            testCategoryList.Add(testcategory);
+            var testAC = await _testRepository.GetTestDetailsByIdAsync(test.Id);
+            await _testRepository.AddSelectedCategoryAsync(testCategoryList);
+            Assert.True(_trappistDbContext.TestCategory.Count() == 1);
+        }
+
+        /// <summary>
+        /// Deselects a category
+        /// </summary>
+        [Fact]
+        public async Task DeselectCategory()
+        {
+            // TODO: When asif's PR is accepted
+            string userName = "niharika@promactinfo.com";
+            ApplicationUser user = new ApplicationUser() { Email = userName, UserName = userName };
+            await _userManager.CreateAsync(user);
+            var category = CreateCategory();
+            await _categoryRepository.AddCategoryAsync(category);
+            var test = CreateTest("Maths");
+            await _testRepository.CreateTestAsync(test);
+            TestCategory testCategory = new TestCategory();
+            testCategory.TestId = test.Id;
+            testCategory.CategoryId = category.Id;
+            List<TestCategory> testCategoryList = new List<TestCategory>();
+            testCategoryList.Add(testCategory);            
+            await _testRepository.AddSelectedCategoryAsync(testCategoryList);
+            var questionAc = CreateQuestionAc(true, "Question in Category", category.Id, 1);           
+            var applicationUser = await _userManager.FindByEmailAsync(user.Email);
+            await _questionRepository.AddSingleMultipleAnswerQuestionAsync(questionAc, applicationUser.Id);
+            TestQuestion testQuestion = new TestQuestion();
+            testQuestion.QuestionId = questionAc.Question.Id;
+            testQuestion.TestId = test.Id;
+            var isExists = await _testRepository.DeselectCategoryAync(category.Id, test.Id);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused 'isExists' local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1481)